File: c++/leetcode/504. Base 7.cpp
class Solution
{
public:
  string convertToBase7(int num)
  {
    if (num == 0)
      return "0";

    vector<int> digits;
    const int base = 7;
    string ans = "";

    ans += num < 0 ? num *= -1, "-" : "";

    cout << num;

    while (num)
    {
      digits.push_back(num % base);
      num /= base;
    }

    for (auto i = digits.rbegin(); i != digits.rend(); ++i)
      ans += to_string(*i);

    return ans;
  }
};

File: javascript/leetcode/504. Base 7.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function (num) {
  if (num == 0) return "0";
  const digits = [];
  const base = 7;
  let ans = "";

  num < 0 ? ((num *= -1), (ans += "-")) : "";

  while (num) {
    digits.push(num % base);
    num = Math.floor(num / base);
  }

  for (let i = digits.length - 1; i >= 0; --i) ans += digits[i];

  return ans;
};
